### ISSUE TITLE
Document propertysource reload file watch option

### ISSUE THREAD

**Issue by ryandawsonuk** [spring-cloud/spring-cloud-kubernetes#190]
The reload options in the [current documentation](https://github.com/spring-cloud-incubator/spring-cloud-kubernetes#propertysource-reload) all require calling the k8s API. This is why we are told that to "access ConfigMaps on OpenShift the service account needs at least view permissions".

But if we mounted the ConfigMap as a volume we could still watch the file for changes. There's an [example of doing this in the JHipster Registry project](https://github.com/jhipster/jhipster-registry/blob/5a0d2354189386a5583381de6486d2f8a6e0130f/src/main/java/io/github/jhipster/registry/service/CloudConfigRefreshService.java#L60) by @srinivasa-vasu. This kind of hot loading wouldn't require any service account. It would be an approach to using configmaps with java that is closer to that with other languages and still allow for hot loading. (See 'Centralized configuration in Kubernetes' section of https://www.jhipster.tech/kubernetes/ for more info.)

This is just a suggestion. I've not actually thought any further than the above for what would be involved in implementing it.

**Comment by spencergibb** [comment]
If it's just watching a file, I don't think that is specific to kubernetes at all.

**Comment by ryandawsonuk** [comment]
Apologies, I should have put more thought into this before raising. Perhaps it should be a Spring Boot or Spring Cloud feature request. I also apologise if my comments seemed critical - this project is providing great tools and options for Spring developers. I am wondering if there's a further option that could be offered. 

Here is a use-case I have in mind. Imagine that a developer is tasked with porting an existing app to Kubernetes. It makes extensive use of '@RefreshScope' but the Architect now wants to use different ConfigMaps for different services. Spring Cloud Kubernetes would be an obvious choice in this scenario. But the cluster admin isn't happy about creating service accounts. (Based loosely on https://stackoverflow.com/questions/50097597/rabbitmq-as-spring-cloud-bus-in-kubernetes-for-spring-boot-applications/50113708?noredirect=1#comment87677210_50113708)

It would save this imaginary developer time and politics if there were a way to watch for config changes without needing a service account. Then they could mount the application.properties to the /config/ directory and use Spring Boot's externalised configuration feature. This would also have the merit of using the volume mount pattern prominent with k8s apps, which is already being prominently used (https://github.com/trisberg/devoxx-spring-boot-k8s/blob/master/actors/config/actors-deployment.yaml) and recommended (https://stackoverflow.com/a/50462990/9705485) for scenarios without use of RefreshScope and which the README for the Spring Cloud Kubernetes project recommends for Secrets (it currently notes "access to secrets via API may be restricted for security reasons, the preferred way is to mount secret to the POD").

Here I am assuming that if I did set up a Spring Boot app to read an externalised application.properties and I were to make a change to that file then there is nothing in the Spring ecosystem currently that I could use to detect that change and automatically trigger a refresh (https://stackoverflow.com/questions/39387272/reload-property-value-when-external-property-file-changes-spring-boot). It seems a solvable problem if the file can be watched and refreshes programmatically triggered. 

Perhaps there is good reason against providing such a feature. Maybe it would encourage overuse of @RefreshScope? Presumably that is why reload is disabled by default in Spring Cloud Kubernetes? Or other disadvantages that I am overlooking?

**Comment by spencergibb** [comment]
Blocked by https://github.com/spring-cloud/spring-cloud-commons/issues/605

**Comment by ryanjbaxter** [comment]
I agree would be good for the next major

**Comment by ryanjbaxter** [comment]
Closing this in favor of https://github.com/spring-cloud/spring-cloud-commons/issues/605
